drivers/timer/intel_adsp_timer.c

```diff
@@ -42,30 +44,36 @@
 BUILD_ASSERT(MIN_DELAY < CYC_PER_TICK);
 BUILD_ASSERT(COMPARATOR_IDX >= 0 && COMPARATOR_IDX <= 1);
 
-#define WCTCS      (ADSP_SHIM_DSPWCTS)
-#define COUNTER_HI (ADSP_SHIM_DSPWCH)
-#define COUNTER_LO (ADSP_SHIM_DSPWCL)
-#define COMPARE_HI (ADSP_SHIM_COMPARE_HI(COMPARATOR_IDX))
-#define COMPARE_LO (ADSP_SHIM_COMPARE_LO(COMPARATOR_IDX))
-
+#define DSP_WCT_CS_TT(x)                     BIT(4 + x)
 
 static struct k_spinlock lock;
 static uint64_t last_count;
 
+/* Not using current syscon driver due to overhead due to MMU support */
+#define SYSCON_REG_ADDR	DT_REG_ADDR(DT_INST_PHANDLE(0, syscon))
```

Comment:
it's not matter of not using it but this is totally broken on MMU platforms (better be explicit about that?).

You are reading / writing from a phys address retrieved from the DT when the actual virt address mapped by the driver could be completely different.

What about (as I suggested before) caching in this driver the value returned by `syscon_get_base` and then using the offsets on that address? The expense would be one single call at init time.

Reply:
>  is totally broken on MMU platforms 

what is totaly broken on MMU?

For this platform I do not care much for MMU. syscon seemed interesting by looking at the description 

> System controller node represents a register region containing a set
of miscellaneous registers. The registers are not cohesive enough to
represent as any specific type of device. The typical use-case is for
some other node's driver, or platform-specific code, to acquire a
reference to the syscon node (e.g. by phandle, node path, or search
using a specific compatible value), interrogate the node (or associated
OS driver) to determine the location of the registers, and access the
registers directly.

Which is basically what I was looking for.

I obviously missed the MMU complexity and additional overhead associated with that and the fact we are dealing with syscalls.

Or am I missing something obvious here? :)

Reply:
> what is totaly broken on MMU?

This driver when (if?) this is going to run on MMU-enabled platforms.

> For this platform I do not care much for MMU

Fair enough.